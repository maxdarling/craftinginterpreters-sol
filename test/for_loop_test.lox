// Test for loop constructs
print "=== For Loop Tests ===";

// Basic for loop
print "Basic for loop:";
for (var i = 0; i < 5; i = i + 1) {
  print i;
}

// For loop with different increment
print "For loop with step 2:";
for (var j = 0; j < 10; j = j + 2) {
  print j;
}

// For loop with decrement
print "For loop decrementing:";
for (var k = 5; k > 0; k = k - 1) {
  print k;
}

// For loop with no initialization
print "For loop with pre-initialized variable:";
var m = 3;
for (; m < 6; m = m + 1) {
  print m;
}

// For loop with no condition (infinite loop with break)
print "For loop with break:";
var n = 0;
for (;;) {
  if (n >= 3) {
    break;
  }
  print n;
  n = n + 1;
}

// For loop with no increment
print "For loop with manual increment:";
for (var p = 0; p < 3;) {
  print p;
  p = p + 1;
}

// Nested for loops
print "Nested for loops:";
for (var outer = 1; outer <= 2; outer = outer + 1) {
  for (var inner = 1; inner <= 2; inner = inner + 1) {
    print "(" + outer + "," + inner + ")";
  }
}

// For loop with complex expressions
print "For loop with complex expressions:";
for (var x = 1; x * x <= 16; x = x + 1) {
  print x + " squared is " + (x * x);
}

// For loop variable scoping
print "For loop variable scoping:";
var outside = "outside";
for (var outside = 0; outside < 2; outside = outside + 1) {
  print "inside: " + outside;
}
print "after loop: " + outside;

print "For loop tests completed";
