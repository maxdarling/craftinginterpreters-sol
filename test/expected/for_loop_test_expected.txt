=== For Loop Tests ===
Basic for loop:
0
1
2
3
4
For loop with step 2:
0
2
4
6
8
For loop decrementing:
5
4
3
2
1
For loop with pre-initialized variable:
3
4
5
For loop with break:
0
1
2
For loop with manual increment:
0
1
2
Nested for loops:
(1,1)
(1,2)
(2,1)
(2,2)
For loop with complex expressions:
1 squared is 1
2 squared is 4
3 squared is 9
4 squared is 16
For loop variable scoping:
inside: 0
inside: 1
after loop: outside
For loop tests completed
